src/Cacher.ts

```diff
@@ -1,7 +1,12 @@
 import Manifest from "./Manifest";
 
 interface Cacher {
-    getManifest(manifestUrl: URL): Promise<Manifest>;
+    getManifest(manifestUrl: URL): Promise<Manifest>;   
+
+    // This should kick off the downloads to enable offline use.
+    enable(): Promise<void>;
+
+    renderStatus(element: HTMLElement): void;
```

Comment:
I think it might be better if `Cacher` is not responsible for view-level code. Instead, I might have have something like this:

```typescript
enum RenderStatus {
    Idle,
    Downloading,
    Downloaded
};

onRenderStatusChange(callback: (status: RenderStatus) => void);
```

If you wanted to communicate some kind of progress percentage (which may not be possible), it would use a discriminated union approach instead:

```typescript
interface Idle {
    kind: "Idle";
}
interface Downloading {
    kind: "Downloading";
    progress: number; // [0, 1]
}

interface Downloaded {
    kind: "Downloaded";
}

type RenderStatus
    = Idle
    | Downloading
    | Downloaded;

onRenderStatusChange(callback: (status: RenderStatus) => void);
```

The callback could then match on `status.kind` and have type-safe access to `status.progress` if and only if `kind === "Downloading"`.